### ISSUE TITLE
iD crashing after trying to save way with more than 2000 nodes / allowed to add more nodes than the maximum of 2000

### ISSUE THREAD

**Issue by MRVDH** [openstreetmap/iD#7677]
I got this error on saving after editing large woodland areas. The problem is that this could have easily been shown in the information panel before saving. What happened after I saw this message was that I tried to split the area into two separate area's. After doing this when I tried to hit Save it gave me the login popup (even though I was logged in), I logged in again and the popup window gave me a 404 not found error. Then I decided to reload the editor because iD saves your unsaved changes (or at least I thought). After reloading most of my changes were gone, which means it stopped saving my unsaved progress after a while apparently. I tried fixing it again and the same errors kept occurring.

TLDR: iD crashed and bugged out after throwing an error on saving that the way contained too many nodes.

Solution: Disable adding nodes to a way after the maximum of 2000? Otherwise at least show a message.

![image](https://user-images.githubusercontent.com/6686385/83967887-903a8780-a8c5-11ea-8ca5-4894d0051755.png)

![image](https://user-images.githubusercontent.com/6686385/83968053-aeed4e00-a8c6-11ea-99ac-0a44fff2399d.png)

![image](https://user-images.githubusercontent.com/6686385/83968063-bf052d80-a8c6-11ea-8a04-4cd0e8d42b89.png)

**Comment by mmd-osm** [comment]
That seems like a duplicate of #7381 and #6030

**Comment by MRVDH** [comment]
Looks like this bug has existed for over a year. Would be great if this can be a high prio one, at least the limitation of the amount of nodes that can be placed. Otherwise I have no way to know how many more/left I can add to an area

**Comment by quincylvania** [comment]
@MRVDH Sorry you lost your edits! This is definitely a bug and I think #7381 and #6030 cover it. I agree we should prioritize a fix—data loss is bad.

**Comment by MapSpot** [comment]
I discovered a workaround that can prevent losing your changes when this occurs. Perform the following:

- From the pop-up window, access openstreetmap.org
- Click Edit to open the iD editor in that window
- Return to your original iD editor window that has all your changes
- Press Esc if the window is frozen or greyed out from trying to save
- Click Save button or Ctrl + S (after making sure to resolve the 2000+ node issue)
- The save menu should open properly now, as long as you keep that second iD window open
